miio/integrations/petfeeder/status.py

```diff
@@ -0,0 +1,85 @@
+from datetime import timedelta
+from enum import Enum
+from typing import Any, Dict
+
+from miio import DeviceStatus
+
+
+class FoodLevel(Enum):
+    Normal = 0
+    Low = 1
+    Empty = 2
+
+
+class FoodOutletStatus(Enum):
+    Ok = 0
+    Blocked = 1
+
+
+class FoodStorageLid(Enum):
+    Closed = 0
+    Open = 1
+
+
+class PetFeederStatus(DeviceStatus):
+    """Container for status reports from the Pet Feeder / Smart Pet Food Dispenser."""
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        """
+        Response from pet feeder (mmgg.feeder.petfeeder):
+
+        {
+            "food_level": 0,
+            "feed_plan": 1,
+            "food_outlet": 0,
+            "undeclared": 0,
+            "clean_days_left": 11,
+            "food_storage_lid": 0,
+            "desiccant_days_left": 21,
+            "weight_level": 0,
+            "wifi_led": 0,
+            "button_lock": 1,
+            "country_code": 255
+        }
+        """
+        self.data = data
+
+    @property
+    def food_level(self) -> FoodLevel:
+        """Current food level / status."""
+        return FoodLevel(self.data["food_level"])
+
+    @property
+    def feed_plan(self) -> bool:
+        """Automatic feeding status."""
+        return bool(self.data["feed_plan"])
+
+    @property
+    def food_outlet(self) -> FoodOutletStatus:
```

Comment:
`is_door_open`?

Reply:
Nop, there was some confusion over the naming used in the first pull.
"food_outlet" is actually a fault status for the chute that the food comes out of, i.e the food outlet.
It's either ok or blocked.

Reply:
How about `food_outlet_blocked` returning a bool?

Reply:
Not something I'd want to do personally.
Feels like a job for a front end. Library provides state of X and front end phrases it.

Functions like "is_on", "is_off" feels wrong to me.

Reply:
I don't want to argue, but in the end it's still a boolean value presenting some state. And if the name gives away it's purpose, it's easier to interpret than a name that is purely based on an internal naming of some external component.

Reply:
I believe it's called a "constructive debate" 😇 
Keeping it vague with something like "state" or "status" feels more natural linguistically and also more general and can be applied to non-bool types.

e.g
{$feature}_state / {$feature}_status

Sounds better than ${feature}_is_on